native/python/pyjp_class.cpp

```diff
@@ -728,8 +728,9 @@ static PyObject *PyJPClass_array(PyJPClass *self, PyObject *item)
 
 	if (self->m_Class == NULL)
 	{
-		PyErr_Format(PyExc_TypeError, "Cannot instantiate unspecified array type");
-		return NULL;
+		PyObject *res = PyObject_CallMethod((PyObject *)self, "__class_getitem__", "O", item);
+		Py_DECREF(item);
+		return res;
```

Comment:
This is the most suspect.  It appears to be using the side effect that JClass never got its javaslot filled in.   I would wonder if one could just set the slot properly and use the regular path.   May not be worth the effort, though it likely deserves a comment on the line to to state what is happening here as I am likely to get confused if I saw this without context.

Reply:
I found this part of the code to be a bit strange. The reason the java slot is not filled in is because JArray is a factory with no underlying java class. I was actually surprised that Python used `__getitem__` here instead of `__class_getitem__` but when I looked into why it is documented in the Python docs for `__getitem__` to take precedence.

Reply:
I'm not going to dig into what exactly is going on but I think it's just a side effect of Jpype's object model. I'm not going to tinker with it and will just leave a comment explaining how we got there.